engine/src/main/java/org/camunda/bpm/engine/ProcessEngineConfiguration.java

```diff
@@ -867,8 +883,9 @@ public int getDefaultNumberOfRetries() {
     return defaultNumberOfRetries;
   }
 
-  public void setDefaultNumberOfRetries(int defaultNumberOfRetries) {
+  public ProcessEngineConfiguration setDefaultNumberOfRetries(int defaultNumberOfRetries) {
```

Comment:
Hmm, you may be right. I had a conversation with Tassilo about this once and had the impression that this would not be a problem.
I can revert this without problems.

Reply:
@ThorbenLindhauer, does this break compatibility? 
According to https://wiki.eclipse.org/Evolving_Java-based_APIs_2 (see Evolving API interfaces - API methods) it does.

Reply:
Yes, it does.

Reply:
You can easily distinguish between `public` and `private` API by looking at the package name: when it contains `.impl.`, the class is not part of the API but implementation and we can change it. If this is not the case (e. g. `package org.camunda.bpm.engine;`) we must ensure binary compatibility. You can look it up here: https://docs.camunda.org/manual/7.11/introduction/public-api/

Reply:
This change was reverted.